include/swift/IDE/TypeCheckCompletionCallback.h

```diff
@@ -80,7 +80,31 @@ Type getPatternMatchType(const constraints::Solution &S, Expr *E);
 /// were type-checked with the code completion expression.
 void getSolutionSpecificVarTypes(
     const constraints::Solution &S,
-    llvm::SmallDenseMap<const VarDecl *, Type> &Result);
+    llvm::SmallDenseMap<VarDecl *, Type> &Result);
+
+/// While this RAII is alive the interface types of the variables defined in
+/// \c SolutionSpecificVarTypes are reset to null types.
+/// Afterwards, their types are reset.
+struct WithSolutionSpecificVarTypesRAII {
+  llvm::SmallVector<VarDecl *> VarsWithSolutionSpecificTypes;
+
+  WithSolutionSpecificVarTypesRAII(llvm::SmallDenseMap<VarDecl *, Type> SolutionSpecificVarTypes) {
+    for (auto SolutionVarType : SolutionSpecificVarTypes) {
+      VarsWithSolutionSpecificTypes.push_back(SolutionVarType.first);
+      if (!SolutionVarType.second->hasArchetype()) {
+        SolutionVarType.first->setInterfaceType(SolutionVarType.second);
+      } else {
+        SolutionVarType.first->setInterfaceType(ErrorType::get(SolutionVarType.second));
+      }
+    }
+  }
+
+  ~WithSolutionSpecificVarTypesRAII() {
+    for (auto Var : VarsWithSolutionSpecificTypes) {
```

Comment:
```suggestion
    for (auto *Var : VarsWithSolutionSpecificTypes) {
```
to clarify this is a pointer.